internal/vcs/git.go

```diff
@@ -30,7 +30,7 @@ type GitRepo struct {
 // managed at storagePath.
 //
 // If storagePath is empty on disk, NewGitRepo initializes GitRepo by cloning
-// the repository. Otherwise, NewGitRepo will do a fetch.
+// the repository. Otherwise, NewGitRepo will do a pull.
```

Comment:
I feel like I don't really understand what the problem with fetching was, especially given the documentation of [git pull](https://git-scm.com/docs/git-pull) saying:

> More precisely, `git pull` runs `git fetch` with the given parameters and then depending on configuration options or command line flags, will call either `git rebase` or `git merge` to reconcile diverging branches.

Given this, it seems like `git fetch && git reset SHA` should've done it, and so I'm not sure `git fetch` was the issue so much as the `git reset SHA` part was.

But either way, as long as the test case reproduced the error prior to the fix, and it's fixed now, we can push this through.

Reply:
I could see the reset being the core problem. Thought about aliasing HEAD to FETCH_HEAD if different but that felt harder to explain and more error prone.

Reply:
Example below, so if git reset HEAD will just reset to what you have locally, you would need to call reset FETCH_HEAD for it to work properly with fetch.
❯ git fetch
❯ git rev-parse HEAD
db656133b17b3b16ae406406089d8c9303c5d595
❯ git rev-parse FETCH_HEAD
954f41122b2fa387ef5e6b2d5845009925d9d169

Running a git pull changes to
❯ git pull
❯ git rev-parse HEAD
954f41122b2fa387ef5e6b2d5845009925d9d169
❯ git rev-parse FETCH_HEAD
954f41122b2fa387ef5e6b2d5845009925d9d169